www/include/options/accessLists/resourcesACL/DB-Func.php

```diff
@@ -319,24 +319,26 @@ function insertLCA()
 {
     global $form, $pearDB;
 
-    $ret = array();
+    $ret = [];
     $ret = $form->getSubmitValues();
-    $rq = "INSERT INTO `acl_resources` ";
-    $rq .= "(acl_res_name, acl_res_alias, all_hosts, all_hostgroups, all_servicegroups, acl_res_activate, " .
-        "changed, acl_res_comment) ";
-    $rq .= "VALUES ('" . $pearDB->escape($ret["acl_res_name"]) . "', " .
-        "'" . $pearDB->escape($ret["acl_res_alias"]) . "', " .
-        "'" . (isset($ret["all_hosts"]["all_hosts"]) ? $pearDB->escape($ret["all_hosts"]["all_hosts"]) : 0) . "', " .
-        "'" . (isset($ret["all_hostgroups"]["all_hostgroups"])
-            ? $pearDB->escape($ret["all_hostgroups"]["all_hostgroups"])
-            : 0) . "', " .
-        "'" . (isset($ret["all_servicegroups"]["all_servicegroups"])
-            ? $pearDB->escape($ret["all_servicegroups"]["all_servicegroups"])
-            : 0) . "', " .
-        "'" . $pearDB->escape(isset($ret["acl_res_activate"]) ? $ret["acl_res_activate"]["acl_res_activate"] : '') . "', " .
-        "'1', " .
-        "'" . $pearDB->escape($ret["acl_res_comment"]) . "')";
-    $dbResult = $pearDB->query($rq);
+
+    $resourceValues = sanitizeResourceParameters($ret);
+
+    $statement = $pearDB->prepare(
+        "INSERT INTO `acl_resources`
+        (acl_res_name, acl_res_alias, all_hosts, all_hostgroups, all_servicegroups, acl_res_activate,
+        changed, acl_res_comment)
+        VALUES (:aclResourceName, :aclResourceAlias, :allHosts, :allHostGroups, :allServiceGroups, :aclResourceActivate,
+        1, :aclResourceComment)"
+    );
+    $statement->bindValue(':aclResourceName', $resourceValues['acl_res_name'], \PDO::PARAM_STR);
+    $statement->bindValue(':aclResourceAlias', $resourceValues['acl_res_alias'], \PDO::PARAM_STR);
```

Comment:
Be careful, a boolean value equal to false (filter_var) will create an empty value with the string format.
Is this the desired behaviour?

Reply:
as discuss with @kduret , acl_res_name is a mandatory field, so if Resource Name is an empty string, an exception will be thrown (13adc79), for alias and comment, an empty string will be inserted as they aren't mandatory fields.